core/src/main/java/org/elasticsearch/index/query/IndexQueryParserService.java

```diff
@@ -80,6 +81,7 @@ protected QueryShardContext initialValue() {
     private boolean queryStringLenient;
     private final ParseFieldMatcher parseFieldMatcher;
     private final boolean defaultAllowUnmappedFields;
+    private Client client;  // NO COMMIT: this is temporary until we figure it out
```

Comment:
I think we may want to use an optional injection and inject through setter instead, similar to what we do in `ScriptService#setClient`